packages/notte-browser/src/notte_browser/window.py

```
@@ -315,18 +315,48 @@ async def snapshot_metadata(self) -> SnapshotMetadata:
+                    f"CDP screenshot for {self.page.url}: attach={attach_ms:.0f}ms send={send_ms:.0f}ms size={len(data)}B"
+                )
+                return data
+            finally:
+                t_detach = time.monotonic()
+                # Shield detach from outer wait_for cancellation — otherwise a timeout here
+                # would skip detach and leak the CDP session, which is the exact contention
+                # this path is trying to surface. Inner wait_for bounds detach itself so a
+                # hung detach can't live forever in the background task set.
+                try:
+                    await asyncio.shield(asyncio.wait_for(cdp_session.detach(), timeout=2.0))
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)